include/RAJA/pattern/kernel/LambdaArgs.hpp

```diff
@@ -0,0 +1,86 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Header file for kernel conditional templates
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC.
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For details about use and distribution, please read RAJA/LICENSE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_pattern_kernel_LambdaArgs_HPP
+#define RAJA_pattern_kernel_LambdaArgs_HPP
+
+
+#include "RAJA/config.hpp"
+
+#include "RAJA/pattern/kernel/internal.hpp"
+
+#include <iostream>
+#include <type_traits>
+
+namespace RAJA
+{
+
+namespace statement
+{
+
+struct seg_t
+{};
+
+struct param_t
+{};
+
+struct offset_t
+{};
+
+template<typename T, camp::idx_t ...>
+struct LambdaArgs
+{
+};
+
+template<camp::idx_t ... Args>
```

Comment:
Why the explicit LambdaArgs instantiations? Aren't these all equivalent to the generic one?